### PR TITLE
iis_iis10_sdh: Accept IIS 10 Log Format

### PR DESCRIPTION
**Pull request by ishleenk17**
- Enhancement

## What does this PR do?

IIS Beats module doesn't support the below type of IIS 10 log format:

2018-12-31 12:02:53 10.44.0.136 GET / - 443 - 10.50.6.188 Mozilla/4.0+(compatible;+MSIE+8.0;+Windows+NT+5.1;+Trident/4.0) - - 401 0 0 0

Gives the below error :

Provided Grok expressions do not match field value: [2021-06-10 23:26:57 10.44.0.136 GET / - 80 - 10.46.208.5 Mozilla/4.0+(compatible;+MSIE+8.0;+Windows+NT+5.1;+Trident/4.0) - itweb.acx.co 200 0 0 23]

`This PR adds support for a log format for IIS 10 logs.` 
A new grok pattern was added that supports parsing cookie in the IIS 10 log. 

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## How to test this PR locally

Pipeline tests have been added for the new format of logs. 
Running elastic-package test pipeline should be good. 

- Closes https://github.com/elastic/beats/issues/31839

### CODE REVIEW

File: packages/iis/data_stream/access/_dev/test/pipeline/tst-iis10-access.log [hunk 1]
```diff
@@ -0,0 +1,3 @@
+2022-05-09 17:10:04 10.119.32.8 POST /civault/Cryptology/Cryptology.svc - 443 - 10.119.0.62 Mozilla/4.0+(compatible;+MSIE+7.0;+Windows+NT+10.0;+Win64;+x64;+Trident/7.0;+.NET4.0C;+.NET4.0E;+.NET+CLR+2.0.50727;+.NET+CLR+3.0.30729;+.NET+CLR+3.5.30729) - [apcvwp00049.corp.acxiom.net](https://apcvwp00049.corp.acxiom.net/) 200 0 0 26
+2022-05-08 01:26:01 10.119.32.8 POST /NamingServiceANSWS/ANSWS.svc - 443 - 10.119.38.250 - - itwebcert.acxiom.com 200 0 0 232
+2021-06-10 23:26:57 10.44.0.136 GET / - 80 - 10.46.208.5 Mozilla/4.0+(compatible;+MSIE+8.0;+Windows+NT+5.1;+Trident/4.0) check_for_cookie itweb.acx.co 200 0 0 23
```
Comment (jsoriano) [inline packages/iis/data_stream/access/_dev/test/pipeline/tst-iis10-access.log:3]: It'd be nice to have real values here.

File: packages/iis/data_stream/access/elasticsearch/ingest_pipeline/default.yml [hunk 2]
```diff
@@ -46,6 +46,11 @@ processors:
           (?:-|%{IPORHOST:source.address}) (?:-|%{NOTSPACE:user_agent.original}) (?:-|%{NUMBER:http.response.status_code:long})
           (?:-|%{NUMBER:iis.access.sub_status:long}) (?:-|%{NUMBER:iis.access.win32_status:long})
           (?:-|%{NUMBER:_temp_.duration:long})( (?:-|%{IPORHOST:network.forwarded_ip}))?'
+        - '%{TIMESTAMP_ISO8601:iis.access.time} (?:-|%{IPORHOST:destination.address}) (?:-|%{WORD:http.request.method})
+          (?:-|%{NOTSPACE:_temp_.url_path}) (?:-|%{NOTSPACE:_temp_.url_query}) (?:-|%{NUMBER:destination.port:long}) (?:-|%{NOTSPACE:user.name})
+          (?:-|%{IPORHOST:source.address}) (?:-|%{NOTSPACE:user_agent.original}) (?:-|%{NOTSPACE:iis.access.cookie}) (?:-|%{NOTSPACE:http.request.referrer})
+          (?:-|%{NUMBER:http.response.status_code:long}) (?:-|%{NUMBER:iis.access.sub_status:long})
+          (?:-|%{NUMBER:iis.access.win32_status:long}) (?:-|%{NUMBER:_temp_.duration:long})( (?:-|%{IPORHOST:network.forwarded_ip}))?'
```
Comment (jsoriano) [inline packages/iis/data_stream/access/elasticsearch/ingest_pipeline/default.yml:53]: At some point we may consider merging some of these patterns.

**Review by jsoriano — APPROVED** [review]
:+1:

File: packages/iis/data_stream/access/_dev/test/pipeline/tst-iis10-access.log (hunk 1 above)
Comment (ishleenk17) [reply packages/iis/data_stream/access/_dev/test/pipeline/tst-iis10-access.log]: Definitely, taken care of!

**Review by mtojek — APPROVED** [review]

**Review by jsoriano — APPROVED** [review]

**Pull request merged**